### PR TITLE
Refactor: remove label and add categories

### PR DESCRIPTION
**Pull request by emilyvera**
Label was only used for the prototype, so I removed that across all files and replaced it with categories instead

### CODE REVIEW

File: src/main/webapp/js/index.js [hunk 1]
```diff
@@ -15,12 +15,12 @@
 
 /** Fetches receipts from the server and adds them to the DOM. */
 async function searchReceipts() {
-  const label = document.getElementById('search-input').value;
-  const response = await fetch(`/search-receipts?label=${label}`);
+  const categories = document.getElementById('category-input').value;
+  const response = await fetch(`/search-receipts?categories=${categories}`);
```
Comment (acheng1) [inline src/main/webapp/js/index.js:19]: Similar comment to your other PR: I would consider using [URLSearchParams](https://developers.google.com/web/updates/2016/01/urlsearchparams) to build the parameter list or use encodeURIComponent to make sure the user input is properly encoded (e.g., will the above won't handle if the input includes a space like "burger king").

File: src/main/java/servlets/SearchServlet.java [hunk 2]
```diff
@@ -47,20 +40,20 @@
...
     response.getWriter().println(gson.toJson(receipts));
   }
 
-  /** Returns ImmutableList of receipts from datastore with the same label as desiredLabel. */
-  private ImmutableList<Receipt> getReceiptsWithMatchingLabel(String desiredLabel) {
-    // Set filter to retrieve only receipts with label equal to desiredLabel.
-    Filter matchingLabels = new FilterPredicate("label", FilterOperator.EQUAL, desiredLabel);
+  /** Returns ImmutableList of receipts from datastore with the same categories. */
+  private ImmutableList<Receipt> getReceiptsWithMatchingCategories(String categories) {
+    // Set filter to retrieve only receipts with categories equal to categories.
```
Comment (acheng1) [inline src/main/java/servlets/SearchServlet.java:53]: Does this mean the user has to remember the exact set of categories they originally entered or that was extracted from the receipt? I would have assumed that if we had the following:

receipt1 - category1, category2, category3
receipt2 - category2
receipt3 - category3
receipt4 - category4

searching for category1 would return receipt1 and searching for category2 would return receipt1 and receipt2. If you searched for category3, category4, the search would return receipt1, receipt3, and receipt4 so would be an OR of the categories the user is searching for.

Comment (emilyvera) [reply src/main/java/servlets/SearchServlet.java:53]: Searching for only one category would return the right receipts. But for the MVP we agreed that the user would only be able to search one category for now

File: src/main/webapp/js/index.js (hunk 1 above)
Comment (emilyvera) [reply src/main/webapp/js/index.js:19]: It handles spaces already

File: src/main/java/servlets/SearchServlet.java [hunk 3]
```diff
@@ -47,20 +40,20 @@
...
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(receipts));
   }
 
-  /** Returns ImmutableList of receipts from datastore with the same label as desiredLabel. */
-  private ImmutableList<Receipt> getReceiptsWithMatchingLabel(String desiredLabel) {
-    // Set filter to retrieve only receipts with label equal to desiredLabel.
-    Filter matchingLabels = new FilterPredicate("label", FilterOperator.EQUAL, desiredLabel);
+  /** Returns ImmutableList of receipts from datastore with the same categories. */
+  private ImmutableList<Receipt> getReceiptsWithMatchingCategories(String categories) {
```
Comment (bryyeh) [inline src/main/java/servlets/SearchServlet.java:52]: If only one category can be searched for this version, can you change all the plural "categories" to be singular.

**Comment by acheng1** [comment]
> Label was only used for the prototype, so I removed that across all files and replaced it with categories instead

Thanks for cleaning this up!

**Pull request merged**